### PR TITLE
Fix generate product command

### PR DESCRIPTION
**Pull request by quisse**
## Subject

I am targeting this branch, because not backwards compatible.

## Changelog

```markdown
### Fixed
- make generate product work again
```

### CODE REVIEW

File: docs/reference/tutorials/create-product.rst [hunk 1]
```diff
@@ -29,24 +29,24 @@ Run the following command to create the files:
 
 .. code-block:: bash
 
-	php bin/console sonata:product:generate Bowl sonata.ecommerce_demo.product.bowl
+	php bin/console sonata:product:generate Bowl sonata.ecommerce_demo.product.bowl App
 
-The required base files will be created in ``src/Sonata/ProductBundle``.
-To finalize the installation, we have to define the missing parameters like the type itself and the related manager. These data have to be provided in ``src/Sonata/ProductBundle/Resources/config/product.yml``.
+The required base files will be created in ``src/Application/Sonata/ProductBundle``.
```
Comment (core23) [inline docs/reference/tutorials/create-product.rst]: We should follow symfony flex style, so no `Application` subfolder please

Comment (quisse) [reply docs/reference/tutorials/create-product.rst]: Got that wrong, fixed it.

File: docs/reference/tutorials/create-product.rst [hunk 2]
```diff
@@ -42,11 +42,11 @@ To finalize the installation, we have to define the missing parameters like the
         sonata.ecommerce_demo.product.bowl.manager:
             class: Sonata\ProductBundle\Entity\ProductManager
             arguments:
-                - App\Sonata\ProductBundle\Entity\Bowl
+                - App\Application\Sonata\ProductBundle\Entity\Bowl
```
Comment (core23) [inline docs/reference/tutorials/create-product.rst]: Revert this please

File: src/ProductBundle/Command/GenerateProductCommand.php [hunk 3]
```diff
@@ -41,9 +52,9 @@ protected function configure(): void
 
     protected function execute(InputInterface $input, OutputInterface $output): void
     {
-        // find a better way to detect the source folder
-        $bundle_dir = sprintf('%s/../src/Sonata/ProductBundle',
-            $this->getContainer()->get('kernel')->getRootDir()
+        // find a better way to detect the Application folder
+        $bundle_dir = sprintf('%s/src/Application/Sonata/ProductBundle',
```
Comment (core23) [inline src/ProductBundle/Command/GenerateProductCommand.php]: Revert this. There is no `Application` folder

File: src/ProductBundle/Command/GenerateProductCommand.php [hunk 4]
```diff
@@ -134,11 +143,11 @@ protected function execute(InputInterface $input, OutputInterface $output): void
     {{ service }}.manager:
         class: Sonata\ProductBundle\Entity\ProductManager
         arguments:
-            - App\Sonata\ProductBundle\Entity\{{ product }}
+            - {{ namespace_prefix }}Application\Sonata\ProductBundle\Entity\{{ product }}
```
Comment (core23) [inline src/ProductBundle/Command/GenerateProductCommand.php]: Revert this. There is no `Application` folder

File: src/ProductBundle/Command/GenerateProductCommand.php [hunk 5]
```diff
@@ -134,11 +143,11 @@ protected function execute(InputInterface $input, OutputInterface $output): void
...
             - "@doctrine"
 
     {{ service }}.type:
-        class: App\Sonata\ProductBundle\Provider\{{ product }}ProductProvider
+        class: {{ namespace_prefix }}Application\Sonata\ProductBundle\Provider\{{ product }}ProductProvider
```
Comment (core23) [inline src/ProductBundle/Command/GenerateProductCommand.php]: Revert this. There is no `Application` folder

File: src/ProductBundle/Command/GenerateProductCommand.php [hunk 6]
```diff
@@ -154,7 +163,7 @@ class: App\Sonata\ProductBundle\Provider\{{ product }}ProductProvider
 
 Add this line in <comment>/src/Sonata/ProductBundle/Resources/config/serializer/Entity.Product.xml</comment>
 
-    <discriminator-class value="{{ service }}">App\Sonata\ProductBundle\Entity\{{ product }}</discriminator-class>
+    <discriminator-class value="{{ service }}">{{ namespace_prefix }}Application\Sonata\ProductBundle\Entity\{{ product }}</discriminator-class>
```
Comment (core23) [inline src/ProductBundle/Command/GenerateProductCommand.php]: Revert this. There is no `Application` folder

File: src/ProductBundle/Resources/skeleton/product/Controller/Controller.php [hunk 7]
```diff
@@ -9,7 +9,7 @@
  * file that was distributed with this source code.
  */
 
-namespace App\Sonata\ProductBundle\Controller;
+namespace {{ namespace_prefix }}Application\Sonata\ProductBundle\Controller;
```
Comment (core23) [inline src/ProductBundle/Resources/skeleton/product/Controller/Controller.php]: Revert this. There is no `Application` folder

File: src/ProductBundle/Resources/skeleton/product/Entity/Entity.php [hunk 8]
```diff
@@ -8,7 +8,7 @@
  * file that was distributed with this source code.
  */
 
-namespace App\Sonata\ProductBundle\Entity;
+namespace {{ namespace_prefix }}Application\Sonata\ProductBundle\Entity;
```
Comment (core23) [inline src/ProductBundle/Resources/skeleton/product/Entity/Entity.php]: Revert this. There is no `Application` folder

File: src/ProductBundle/Resources/skeleton/product/Provider/EntityProductProvider.php [hunk 9]
```diff
@@ -8,7 +8,7 @@
  * file that was distributed with this source code.
  */
 
-namespace App\Sonata\ProductBundle\Provider;
+namespace {{ namespace_prefix }}Application\Sonata\ProductBundle\Provider;
```
Comment (core23) [inline src/ProductBundle/Resources/skeleton/product/Provider/EntityProductProvider.php]: Revert this. There is no `Application` folder

File: src/ProductBundle/Resources/skeleton/product/Repository/Repository.php [hunk 10]
```diff
@@ -8,7 +8,7 @@
  * file that was distributed with this source code.
  */
 
-namespace App\Sonata\ProductBundle\Repository;
+namespace {{ namespace_prefix }}Application\Sonata\ProductBundle\Repository;
```
Comment (core23) [inline src/ProductBundle/Resources/skeleton/product/Repository/Repository.php]: Revert this. There is no `Application` folder

File: src/ProductBundle/Resources/skeleton/product/Resources/config/doctrine/Entity.orm.xml [hunk 11]
```diff
@@ -8,7 +8,7 @@
             xml mapping          : http://www.doctrine-project.org/projects/orm/2.0/docs/reference/xml-mapping/en
             association mapping  : http://www.doctrine-project.org/projects/orm/2.0/docs/reference/association-mapping/en
     -->
-    <entity name="App\Sonata\ProductBundle\Entity\{{ product }}" table="product__product" repository-class="App\Sonata\ProductBundle\Repository\{{ product }}Repository">
+    <entity name="{{ namespace_prefix }}Application\Sonata\ProductBundle\Entity\{{ product }}" table="product__product" repository-class="{{ namespace_prefix }}Application\Sonata\ProductBundle\Repository\{{ product }}Repository">
```
Comment (core23) [inline src/ProductBundle/Resources/skeleton/product/Resources/config/doctrine/Entity.orm.xml]: Revert this. There is no `Application` folder

File: src/ProductBundle/Resources/skeleton/product/Resources/config/serializer/Entity.xml [hunk 12]
```diff
@@ -3,7 +3,7 @@
     <!--
          This file has been generated by the Sonata Product generator.
     -->
-    <class name="App\Sonata\ProductBundle\Entity\{{ product }}" exclusion-policy="all" xml-root-name="{{ root_name }}" discriminator-field-name="type">
+    <class name="{{ namespace_prefix }}Application\Sonata\ProductBundle\Entity\{{ product }}" exclusion-policy="all" xml-root-name="{{ root_name }}" discriminator-field-name="type">
```
Comment (core23) [inline src/ProductBundle/Resources/skeleton/product/Resources/config/serializer/Entity.xml]: Revert this. There is no `Application` folder

**Comment by quisse** [comment]
@core23 nvm the review for now, seems that i haven't applied the previous requested changes.

**Comment by core23** [comment]
Thanks @quisse

**Pull request merged**